docs/proposals/images/developer/diagram.plantuml

```diff
@@ -0,0 +1,81 @@
+@startuml
+actor Client as C
+participant "API Server" as A
+participant "Core Manager" as core
+participant "Infrastructure Manager" as infra
+participant "Bootstrap Manager" as boot
+
+core --> A: watch for machines and clusters
+infra --> A: watch for InfraMachines, InfraClusters, Machines, and Clusters
+boot --> A: watch for BootstrapConfigs and Machines
+
+C -> A: Create a Cluster
+A -> core: Cluster has been created
+activate core
+note right of core
+    Cluster Controller
+endnote
+core -> core: Requeue until Cluster is marked as ready
```

Comment:
```suggestion
core -> core: Requeue until InfrastructureRef is set and InfraCluster is marked as ready
```

Reply:
This seems like too much detail 🤔 Do you feel this is not useful without referencing the field names or implementation details?

Reply:
Maybe I'm getting stuck because I know the details, but the current verbiage is confusing since we are saying to requeue until the Cluster is ready, but the Cluster will not be ready until we set it as ready below.

Reply:
FWIW, I'd lean on being more explicit than leaving doubts for a reader

Reply:
yeah, but there's definitely a tipping point where adding too much detail to remove doubt causes confusion. I'm looking for concepts generally. CAPI cares about when a Cluster is ready to be used. Yes, that means InfrastructureCluster is marked as ready and updated in etcd, but i don't think that level of detail is necessary. I would like to find a balance though, just having trouble understanding the confusion with the words. Maybe we can do a zoom tomorrow and figure it out because this kind of thing is just Hard over text.

Reply:
I think I found a good middle ground. This should read `InfraCluster` and I think that will clear things up.